Samples/Client/SwiftClient/SwiftClient/VideoStreamViewController.swift

```diff
@@ -0,0 +1,642 @@
+    var otherPeers = [Int: String]()
+    var myId: Int!
+    var heartBeatTimer: Timer!
+    var heartBeatTimerIsRunning = false
+    let heartbeatIntervalInSecs = 5
+    let server = Config.server
+    let localName = "ios_client"
+    var messageCounter = 0
+    var peerConnection: RTCPeerConnection!
+    var peerConnectionFactory = RTCPeerConnectionFactory()
+    var newIceCandidatePeerId = -1
+    var peerListDisplay = ["Select"]
+    var hangingGet: URLSessionTask!
+    var request: URLSessionDataTask!
+    @IBOutlet weak var renderView: RTCEAGLVideoView!
+    var videoTrack: RTCVideoTrack!
+    var videoStream: RTCMediaStream!
+    var inputChannel: RTCDataChannel!
+    lazy var navTransform: [CGFloat] = {
+        [unowned self] in
+        self.matCreate()
+        }()
+    var navHeading: CGFloat = 0.0
+    var navPitch: CGFloat = 0.0
+    var navLocation: [CGFloat] = [ 0.0, 0.0, 0.0 ]
+    var isFingerDown = false
+    var fingerDownX: CGFloat = 0
+    var fingerDownY: CGFloat = 0
+    var downPitch: CGFloat = 0.0
+    var downHeading: CGFloat = 0.0
+    var downLocation: [CGFloat] = [ 0.0, 0.0, 0.0 ]
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        renderView.delegate = self
+        
+        pickerView.dataSource = self
+        pickerView.delegate = self
+        
+        initGestureRecognizer()
+        connect()
+    }
+    
+    func joinPeer(peerId: Int) {
+        createPeerConnection(peerId: peerId)
+        inputChannel = self.peerConnection.dataChannel(forLabel: "inputDataChannel", configuration: RTCDataChannelConfiguration())
+        inputChannel.delegate = self
+        
+        let offerOptions = RTCMediaConstraints(mandatoryConstraints: ["OfferToReceiveAudio": "false","OfferToReceiveVideo": "true"], optionalConstraints: nil)
+        
+        peerConnection.offer(for: offerOptions) { (sdp, error) in
+            let sdpH264 = sdp!.sdp.replacingOccurrences(of: "96 98 100 102", with: "100 96 98 102")
+            let receivedOffer = RTCSessionDescription(type: .offer, sdp: sdpH264)
+            self.peerConnection.setLocalDescription(receivedOffer, completionHandler: { (error) in
+                
+                let jsonData = [
+                    "type": "offer",
+                    "sdp": receivedOffer.sdp
+                ]
+                
+                self.sendToPeer(peerId: peerId, data: self.convertDictionaryToData(dict: jsonData)!)
+            })
+        }
+    }
+    
+    func createPeerConnection(peerId: Int) {
+        newIceCandidatePeerId = peerId
+        
+        let defaultPeerConnectionConstraints = RTCMediaConstraints(mandatoryConstraints: nil,
+                                                                   optionalConstraints: ["DtlsSrtpKeyAgreement": "true"])
+        let rtcConfig = RTCConfiguration()
+        let iceServer = RTCIceServer(urlStrings: ["turn:turnserver3dstreaming.centralus.cloudapp.azure.com:5349"],
```

Comment:
nit: these can't be checked in. 😭 we should go roll these creds now. in the future, we need to drive this stuff from some sort of configuration file, or user input